node/node.go

```diff
@@ -244,7 +248,16 @@ func New(cfg *Config, version string, logLevel *utils.LogLevel) (*Node, error) {
 		if synchronizer != nil {
 			syncReader = synchronizer
 		}
-		rpcHandler = rpc.New(chain, syncReader, throttledVM, version, log, &cfg.Network).WithGateway(gatewayClient).WithFeeder(client)
+
+		submittedTransactionsCache := rpccore.NewSubmittedTransactionsCache(
+			int(cfg.SubmittedTransactionsCacheSize),
+			cfg.SubmittedTransactionsCacheEntryTTL,
+		)
+
+		rpcHandler = rpc.New(chain, syncReader, throttledVM, version, log, &cfg.Network).
+			WithGateway(gatewayClient).
+			WithFeeder(client).
+			WithSubmittedTransactionsCache(submittedTransactionsCache)
```

Comment:
Maybe this is not a problem of this PR, but we're mixing between a constructor and builder pattern here.

Reply:
Do you find something inherently wrong with it?

The way I see it that we have now I think is:
1. This hybrid approach where we have mandatory fields and optional fields are set through the builder pattern
2. Do you think we should go full constructor or full build pattern? I think it makes code harder to read, but happy to hear your thoughts

Reply:
I think there are 3 kinds of fields here:
- Mandatory fields like `chain` or `syncReader`. Without them, nothing can work.
- Optional fields like `submittedTransactionsCache`. With the current implementation, everything works fine with or without it.
- Fields like gateway and feeder are unclear to me. They're currently optional by builder pattern, but at the same time we throw 500 if they're missing in some particular endpoints. At the same time, `vm` is mandatory, but it's similar to gateway and feeder because it's only used in some particular endpoints like trace and simulate.

Reply:
I don't have strong opinion here, because probably the gain is probably not much, but I think:
- If we prefer builder pattern, there should be a `Build` function to maintain immutability. Reason: we can call `With...` 20 lines below, which make it error prone and hard to debug/reason.
- Another approach which is more idiomatic in my opinion is Option pattern. In short, Option is a function that mutate the implementation type (private struct). Options are used as variadic arguments of the constructor, which return the interface type (public interface).